### PR TITLE
Restyle release list, fix branch dropdown

### LINKED ISSUE #30821: ui bug on mobile on releases page
### Description

image

on pixel 8 pro

### Gitea Version

trunk

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

docker

### Database

None

### PR DESCRIPTION
**Pull request by silverwind**
Fixes https://github.com/go-gitea/gitea/issues/30821 and restyles the release list. 

Desktop:

Screenshot 2024-05-02 at 20 46 10

Mobile:

Screenshot 2024-05-02 at 20 46 21

### CODE REVIEW

File: web_src/css/repo/release-tag.css [hunk 1]
```diff
@@ -13,58 +14,81 @@
...
-  left: -5.5px;
-  top: 30px;
-  border-radius: var(--border-radius-circle);
-  border: 2.5px solid var(--color-body);
+#release-list .release-entry .detail .download .list li:last-child {
+  border-bottom: none;
 }
 
-.repository.tags #tags-table .tag {
+#tags-table .tag {
```
Comment (wxiaoguang) [inline web_src/css/repo/release-tag.css]: Not a blocker, but I think it is a good chance to rename these one-word name `.meta` / `.tag` to some longer names, to make them easier to search from tmpl code.

**Review by wxiaoguang — APPROVED** [review]

**Comment by lunny** [comment]
Sorry, I like the previous version better. I personally don't like too many frames.

**Comment by silverwind** [comment]
> Sorry, I like the previous version better. I personally don't like too many frames.

I think you will get used to it, it's pretty close to what github has now.

File: web_src/css/repo/release-tag.css (hunk 1 above)
Comment (silverwind) [reply web_src/css/repo/release-tag.css]: Hmm yes, could be `tag-entry` or `tag-row` maybe.

**Review by lunny — APPROVED** [review]

**Comment by wxiaoguang** [comment]
> Hmm yes, could be `tag-entry` or `tag-row` maybe.


Would you like to do it in this PR or defer it? I would prefer to do it together since this change is being tested.

**Comment by silverwind** [comment]
Yes I can do it here

**Comment by yp05327** [comment]
I like the previous version too. or at least the background color should be same.

**Comment by silverwind** [comment]
Maybe these boxes need a bit less color contrast against the page background. Would that work for you guys?

**Comment by yp05327** [comment]
Update my comment above:
For the PC view, I prefer the old design, but for the Mobile view, the new design looks good.

**Comment by silverwind** [comment]
@wxiaoguang renamed `tag` to `tag-list-row`. I couldn't find any more matches for this class in JS and CSS.

As for the design, I think the pro's outweight the cons and I find the previous design without a box looking dated. Imho we should "boxify" more things like the issue list in the future.

**Pull request merged**